### PR TITLE
chore: bump rust to 1.71.0

### PR DESCRIPTION
**Pull request by iGxnon**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

    Refer to https://github.com/rust-lang/rust/issues/112430.
    When switching branches or writing mockall tests, we may encounter bugs in rustc's incremental compilation. The specific reason is unknown, but updating rustc and mockall versions can resolve it.

* what changes does this pull request make?

   Bump rust to 1.71.0

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

   Exposing APIs to users will not, but for developers, they need to update their own rustc.

### CODE REVIEW

File: workspace-hack/Cargo.toml [hunk 1]
```diff
@@ -26,7 +26,8 @@ getrandom = { version = "0.2", default-features = false, features = ["js", "rdra
 hashbrown = { version = "0.14", default-features = false, features = ["raw"] }
 hyper = { version = "0.14", features = ["full"] }
 indexmap = { version = "1", default-features = false, features = ["std"] }
-itertools = { version = "0.10" }
+itertools-93f6ce9d446188ac = { package = "itertools", version = "0.10" }
+itertools-a6292c17cd707f01 = { package = "itertools", version = "0.11" }
```
Comment (Phoenix500526) [inline workspace-hack/Cargo.toml:30]: Can we unify these two itertools versions?

File: benchmark/src/runner.rs [hunk 2]
```diff
@@ -82,6 +82,7 @@ impl Stats {
     /// Calculate the histogram from the latencies.
     #[allow(clippy::indexing_slicing)]
     #[allow(clippy::let_underscore_must_use)] // the writeln! will not fail thus always return an Ok
+    #[allow(clippy::arithmetic_side_effects)] // test only
```
Comment (Phoenix500526) [inline benchmark/src/runner.rs:85]: Should the comment here be "benchmark only"?

File: workspace-hack/Cargo.toml (hunk 1 above)
Comment (iGxnon) [reply workspace-hack/Cargo.toml:30]: Probably not, this is generated by Hakari.

Comment (iGxnon) [reply workspace-hack/Cargo.toml:30]: 0.10 is what we are using, 0.11 is used by some dependent libraries. Do we need to update the itertools version?
Its latest version is 0.12.

File: benchmark/src/runner.rs (hunk 2 above)
Comment (iGxnon) [reply benchmark/src/runner.rs]: done

File: workspace-hack/Cargo.toml (hunk 1 above)
Comment (iGxnon) [reply workspace-hack/Cargo.toml]: updated to 0.11

**Review by bsbds — APPROVED** [review]

**Review by bsbds — APPROVED** [review]

**Pull request merged**